File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.cpp
#include <string>
#include <vector>
#include <map>
#include <algorithm>

using namespace std;

int solution(int k, vector<int> tangerine) {
    int answer = 0;
    
    map<int,int> m;
    for(int i = 0; i<tangerine.size(); i++){
        m[tangerine[i]]++;
    }
    
    vector<int> v;
    for(auto iter = m.begin(); iter!=m.end(); iter++){
        v.push_back(iter->second);
    }
    sort(v.begin(),v.end(),greater<int>());
    
    int cnt = 0;
    for(int i = 0; i<v.size(); i++){
        cnt += v[i];
        if(cnt >= k){
            answer = i + 1;
            break;
        }
    }
    
    return answer;
}

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.java
import java.util.*;

class Solution {
    public int solution(int k, int[] tangerine) {
        int answer = 0;
        
        HashMap<Integer, Integer> map = new HashMap<>();
        for(int i : tangerine) {
            map.put(i, map.getOrDefault(i, 0) + 1);
        }
        
        List<Integer> key = new ArrayList<>(map.keySet());
        Collections.sort(key, new Comparator<Integer>(){
           public int compare(Integer o1, Integer o2) {
               return map.get(o2) - map.get(o1);
           } 
        });
        
        int cnt = 0;
        for(int x : key) {
            cnt += map.get(x);
            answer++;
            if(cnt >= k) break;
        }
        
        return answer;
    }
}